contracts/01-tamagotchi/tests/smoke.rs

```diff
@@ -1,10 +1,39 @@
-use gtest::{Program, System};
+use gtest::{Log, Program, System};
+use tamagotchi_io::*;
 
 #[test]
 fn smoke_test() {
     let sys = System::new();
     sys.init_logger();
-    let _program = Program::current(&sys);
+    let program = Program::current(&sys);
+    let result = program.send(2, String::from("Goodmoring"));
+    assert!(!result.main_failed());
+    let result = program.send(2, TmgAction::Name);
+    let log = Log::builder()
+        .dest(2)
+        .payload(TmgEvent::Name(String::from("Goodmoring")));
+    assert!(result.contains(&log));
+    let _result = program.send(2, TmgAction::Age);
+    // let log = Log::builder().dest(2).payload(TmgEvent::Age(sys.block_timestamp()));
+    // assert!(result.contains(&log));
```

Comment:
You can use [`System::spend_blocks`](https://docs.gear.rs/gtest/struct.System.html#method.spend_blocks) to go forward and emulate the changed timestamp.